src/python/grpcio/grpc/experimental/aio/_call.py

```diff
@@ -56,207 +81,335 @@ def __init__(self,
+            return _OK_CALL_REPRESENTATION.format(
+                self.__class__.__name__, self._code,
+                self._status.result().self._status.result().details())
+        else:
+            return _NON_OK_CALL_REPRESENTATION.format(
+                self.__class__.__name__, self._code,
+                self._status.result().details(),
+                self._status.result().debug_error_string())
+
+    def __repr__(self) -> str:
+        return self._repr()
+
+    def __str__(self) -> str:
+        return self._repr()
+
+
+class UnaryUnaryCall(Call, _base_call.UnaryUnaryCall):
+    """Object for managing unary-unary RPC calls.
+
+    Returned when an instance of `UnaryUnaryMultiCallable` object is called.
+    """
+    _loop: asyncio.AbstractEventLoop
+    _request: RequestType
+    _deadline: Optional[float]
+    _channel: cygrpc.AioChannel
+    _method: bytes
+    _request_serializer: SerializingFunction
+    _response_deserializer: DeserializingFunction
+    _call: asyncio.Task
 
-        return self._details
+    def __init__(self, request: RequestType, deadline: Optional[float],
+                 channel: cygrpc.AioChannel, method: bytes,
+                 request_serializer: SerializingFunction,
+                 response_deserializer: DeserializingFunction) -> None:
+        super().__init__()
+        self._loop = asyncio.get_event_loop()
+        self._request = request
+        self._deadline = deadline
+        self._channel = channel
+        self._method = method
+        self._request_serializer = request_serializer
+        self._response_deserializer = response_deserializer
+        self._call = self._loop.create_task(self._invoke())
+
+    def __del__(self) -> None:
+        if not self._call.done():
+            self._cancel(
+                cygrpc.AioRpcStatus(cygrpc.StatusCode.cancelled,
+                                    _GC_CANCELLATION_DETAILS, None, None))
+
+    async def _invoke(self) -> ResponseType:
+        serialized_request = _common.serialize(self._request,
+                                               self._request_serializer)
+
+        # NOTE(lidiz) asyncio.CancelledError is not a good transport for
+        # status, since the Task class do not cache the exact
+        # asyncio.CancelledError object. So, the solution is catching the error
+        # in Cython layer, then cancel the RPC and update the status, finally
+        # re-raise the CancelledError.
+        serialized_response = await self._channel.unary_unary(
+            self._method,
+            serialized_request,
+            self._deadline,
+            self._cancellation_future,
+            self._set_initial_metadata,
+            self._set_status,
+        )
+        await self._raise_rpc_error_if_not_ok()
+
+        return _common.deserialize(serialized_response,
+                                   self._response_deserializer)
+
+    def _cancel(self, status: cygrpc.AioRpcStatus) -> bool:
```

Comment:
Seems that we are using the same object type for the cancellation and the status, IMO would be nice if we could use an ad-hoc method for cancellation, for example, `AioRpcCancellationStatus` with only two attributes the desired code and the details.

Reply:
But here you are using the `AioRpcStatus` for notifying to the Cython layer how you would like to cancel the RPC, this would be used later for the Cython layer for canceling the RPC and for building the final `AioRpcStatus`.

I'm still thinking that both use cases are different and would need different classes.

In any case, Im not block this PR for this small thing.